### PR TITLE
Colony Drop (code only)

### PR DESCRIPTION
**Pull request by GunHog**
Adds the backbone for @KorPhaeron's colony drop feature, without the map changes. This will allow @WJohn to create a map change to his standards.

There are still some changes I can probably make to this code though.

To Do:
- [x] Find some way to check that the drop zone is fully within only one area (so as not to breach ruins / necropolis)
- [x] Have a confirmation prompt for the targeter, OR allow the targeter to be used multiple times until the colony itself is launched. **EDIT:** This is new a do_after

Original PR: https://github.com/tgstation/tgstation/pull/19602

### CODE REVIEW

File: code/modules/mining/mine_items.dm [hunk 1]
```diff
@@ -500,3 +500,72 @@
...
+	shuttle_id = "colony_drop"
+	dwidth = 11
+	dheight = 0
+	width = 23
+	height = 23
+	lz_dir = 1
+
+/obj/item/device/assault_pod/mining/attack_self(mob/living/user)
+	var/turf/T = get_turf(user)
+	if(T.z != ZLEVEL_MINING)
```
Comment (Xhuis) [inline code/modules/mining/mine_items.dm]: You should totally add an admin override variable so players could trade their TC to drop a base on the station.

Comment (GunHog) [inline code/modules/mining/mine_items.dm]: If I can work out the best way to handle that (I am thinking of giving it a prompt or a menu for confirmation), I would be more than happy to do that!

File: code/modules/mining/mine_items.dm [hunk 2]
```diff
@@ -500,3 +500,72 @@
...
+	..()
+
+/obj/item/device/assault_pod/mining
+	name = "Landing Field Designator"
+	icon_state = "gangtool-purple"
+	item_state = "electronic"
+	icon = 'icons/obj/device.dmi'
+	desc = "Deploy to designate the landing zone of the auxillary base."
+	shuttle_id = "colony_drop"
+	dwidth = 11
```
Comment (GunHog) [inline code/modules/mining/mine_items.dm]: I need to know how to allow the mappers to change these dimensions, so they can size their colonies to best fit their own maps.

Comment (KorPhaeron) [inline code/modules/mining/mine_items.dm]: edit vars on the targeter

**Comment by GunHog** [comment]
There is now a badmin version for you, Xhuis. Also, I added a way to check that the colony does not intersect more than one area - so you have to be completely inside a sufficiently large ruin if you want to drop a base there.

Obviously the mapped colonies will not be as ROFLHUGE as the original test merge one was.

**Comment by Dorsisdwarf** [comment]
Does this let me set any area to be colony-dropped, so I can badmin colony-drop shitcurity into the gulag?

**Comment by KorPhaeron** [comment]
no

**Comment by GunHog** [comment]
New in this commit:
- Adds Do_After as comfirmation of setting the colony's location.
- Adds an announcement for when the drop zone is set.

**Comment by optimumtact** [comment]
is this ready?

**Comment by GunHog** [comment]
Yes. Any additional adjustments I can do once @WJohn sends me his map update.

**Pull request merged**